src/net_processing.cpp

```diff
@@ -233,6 +240,11 @@ struct Peer {
     std::atomic_bool m_wants_addrv2{false};
     /** Whether this peer has already sent us a getaddr message. */
     bool m_getaddr_recvd{false};
+    /** Number of addr messages that can be processed from this peer. Start at 1 to
+     *  permit self-announcement. */
+    double m_addr_token_bucket{1.0};
```

Comment:
It looks like the code makes sure that this number never leaves the range `0.0..MAX_ADDR_PROCESSING_TOKEN_BUCKET`, right?
(I generally don't like using floating point for countable objects because of potential infinite loops when the value becomes too large or negative to be represented with a granularity of 1, but I think this issue is avoided)

Reply:
It could exceed that limit - when we send `getaddr` we increment it with `MAX_ADDR_TO_SEND` (1000) regardless of its current value. It should be ok wrt concerns about floating point, given that we send `getaddr` just once.

Reply:
> It looks like the code makes sure that this number never leaves the range 0.0..MAX_ADDR_PROCESSING_TOKEN_BUCKET, right?

Not quite. `m_addr_token_bucket` may exceed `MAX_ADDR_PROCESSING_TOKEN_BUCKET` immediately after we send a `getaddr` message. I think that the theoretic maximum it can reach is 1007 (we send the `getaddr` as soon as we receive the peer's `version` message, which can be up to 60 seconds after the connection is established).

Reply:
Right, thanks, that's not really going to give problems.